src/test/regress/specs/isolation_progress_monitoring.spec

```diff
@@ -19,6 +19,22 @@ setup
 	RETURNS TABLE(step int, progress bigint)
 	AS 'citus'
 	LANGUAGE C STRICT;
+
+	CREATE FUNCTION sample_operation(command_type bigint, lockid bigint, progress bigint)
+		RETURNS VOID AS $$
+	BEGIN
+		PERFORM create_progress(command_type, 2);
+		PERFORM pg_advisory_lock(lockid);
+		PERFORM update_progress(0, progress);
+		PERFORM pg_advisory_lock(lockid+1);
+		PERFORM update_progress(1, progress);
+		PERFORM pg_advisory_lock(lockid+2);
+		PERFORM finish_progress();
+		PERFORM pg_advisory_unlock(lockid);
+		PERFORM pg_advisory_unlock(lockid+1);
+		PERFORM pg_advisory_unlock(lockid+2);
```

Comment:
Yeah, that's a lot of locks but to be able to control each step, I added these locks. Not sure if it's possible to do with less locks. 

I added some comments about the rationale of the test scenario and also converted the ones inside the `sample_operation` function to `pg_advisory_xact_lock`s.